### PR TITLE
Jetpack dashboard navigation: add external link icon to Plans link

### PR DESCRIPTION
**Pull request by nunyvega**
Fixes #

## Proposed changes:

Add external link icon to Plans link because this link takes you to Jetpack Cloud

**Before:** 
Screenshot 2023-02-02 at 10 25 11

**After:** 
Screenshot 2023-02-02 at 10 24 02

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to Jetpack Dashboard
* Confirm that Plans in the navigation has the external link icon

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
Would it be an option to use the `ExternalLink` component in the Nav component?

**Comment by nunyvega** [comment]
> Would it be an option to use the `ExternalLink` component in the Nav component?

Thanks for the recommendation @jeherve! I added the ExternalLink component.

File: projects/plugins/jetpack/_inc/client/components/section-nav/item.jsx [hunk 1]
```diff
@@ -30,33 +31,46 @@ class NavItem extends React.PureComponent {
...
-					disabled={ this.props.disabled }
-					role="menuitem"
-					rel={ this.props.isExternalLink ? 'external' : null }
-				>
-					<span className={ 'dops-section-nav-' + itemClassPrefix + '__text' }>
-						{ this.props.children }
-						{ 'number' === typeof this.props.count && <Count count={ this.props.count } /> }
-					</span>
-				</a>
+				{ ! this.props.isExternalLink && (
```
Comment (retrofox) [inline projects/plugins/jetpack/_inc/client/components/section-nav/item.jsx:42]: minor suggestion: You could use ternary condition:

```es6
{ ! this.props.isExternalLink
  ? ( ... )
  : ( ... )
}
```

**Review by retrofox — APPROVED** [review]
LGTM.

File: projects/plugins/jetpack/_inc/client/components/section-nav/item.jsx (hunk 1 above)
Comment (nunyvega) [reply projects/plugins/jetpack/_inc/client/components/section-nav/item.jsx:42]: thanks!

**Pull request merged**